UnknownHostException for us-east-1

When using a us-east-1 bucket, the process fails to bootstrap with following exception:

`Exception in thread "main" com.amazonaws.SdkClientException: Unable to execute HTTP request: yelp-service-data-us-east-1.s3.US.amazonaws.com
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleRetryableException(AmazonHttpClient.java:1175)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1121)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5052)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4998)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4992)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:895)
	at com.yelp.nrtsearch.server.utils.ArchiverImpl.getResources(ArchiverImpl.java:128)
	at com.yelp.nrtsearch.server.luceneserver.RestoreStateHandler.restore(RestoreStateHandler.java:48)
	at com.yelp.nrtsearch.server.grpc.LuceneServer.start(LuceneServer.java:179)
	at com.yelp.nrtsearch.server.grpc.LuceneServer.main(LuceneServer.java:212)
Caused by: java.net.UnknownHostException: yelp-service-data-us-east-1.s3.US.amazonaws.com
	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:800)
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1495)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1354)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1288)
	at com.amazonaws.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:27)
	at com.amazonaws.http.DelegatingDnsResolver.resolve(DelegatingDnsResolver.java:38)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:112)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at com.amazonaws.http.conn.ClientConnectionManagerFactory$Handler.invoke(ClientConnectionManagerFactory.java:76)
	at com.amazonaws.http.conn.$Proxy11.connect(Unknown Source)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at com.amazonaws.http.apache.client.impl.SdkHttpClient.execute(SdkHttpClient.java:72)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1297)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113)`

--- a/src/main/java/com/yelp/nrtsearch/LuceneServerModule.java
+++ b/src/main/java/com/yelp/nrtsearch/LuceneServerModule.java
@@ -83,7 +83,7 @@ public class LuceneServerModule extends AbstractModule {
       // https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/Region.html#US_Standard
       // However, this causes an UnknownHostException so we override it to the full region name
       if (region.equals("US")) {
-          region = "us-east-1";
+        region = "us-east-1";
       }
       String serviceEndpoint = String.format("s3.%s.amazonaws.com", region);
       return AmazonS3ClientBuilder.standard()

--- a/src/main/java/com/yelp/nrtsearch/LuceneServerModule.java
+++ b/src/main/java/com/yelp/nrtsearch/LuceneServerModule.java
@@ -79,6 +79,12 @@ public class LuceneServerModule extends AbstractModule {
       AmazonS3 s3ClientInterim =
           AmazonS3ClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
       String region = s3ClientInterim.getBucketLocation(luceneServerConfiguration.getBucketName());
+      // In useast-1, the region is returned as "US" which is an equivalent to "us-east-1"
+      // https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/Region.html#US_Standard
+      // However, this causes an UnknownHostException so we override it to the full region name
+      if (region.equals("US")) {
+          region = "us-east-1";
+      }
       String serviceEndpoint = String.format("s3.%s.amazonaws.com", region);
       return AmazonS3ClientBuilder.standard()
           .withCredentials(awsCredentialsProvider)

